docs/api-reference.md

```diff
@@ -3,25 +3,623 @@ id: api-reference
+| upsertOption     | A boolean value that defines whether to create a new item if an existing item meeting the specified criteria was not found. |
+
+#### Example
+
+```js
+[STORY_UPVOTED]: async (store, { aggregateId, payload: { userId } }) => {
+  const story = await store.findOne(
+    'Stories',
+    { id: aggregateId },
+    { votes: 1 }
+  )
+  await store.update(
+    'Stories',
+    { id: aggregateId },
+    { $set: { votes: story.votes.concat(userId) } }
+  )
+},
+```
+
+### delete
+
+Deletes data items based on the specified search expression.
+
+#### Arguments
+
+| Argument Name    | Description                                 |
+| ---------------- | ------------------------------------------- |
+| tableName        | A table name.                               |
+| searchExpression | An object that defines a search expression. |
+
+#### Example
+
+```js
+[SHOPPING_LIST_REMOVED]: async (store, { aggregateId }) => {
+  await store.delete('ShoppingLists', { id: aggregateId })
+}
+```
+
+The code sample below demonstrates how to use this API to communicate with a store from a Read Model projection and resolver.
+
+## Saga API
+
+A saga's event handler receives an object that provides access to the saga-related API. This API includes the following objects:
+
+| Object Name | Description                                                                       |
+| ----------- | --------------------------------------------------------------------------------- |
+| store       | Provides access to the saga's persistent store (similar to the Read Model store). |
+| sideEffects | Provides access to the saga's side effect functions.                              |
+
+In addition to user-defined side effect functions, the `SideEffects` object contains the following default side effects:
+
+| Function Name                       | Description                                                                                 |
+| ----------------------------------- | ------------------------------------------------------------------------------------------- |
+| [executeCommand](#executecommand)   | Sends a command with the specified payload to an aggregate.                                 |
+| [scheduleCommand](#schedulecommand) | Similar to `executeCommand`, but delays command execution until a specified moment in time. |
+
+The `sideEffects` object's `isEnabled` field indicates whether or not side effects are enabled for the saga.
+
+### executeCommand
+
+Sends a command with the specified payload to an aggregate.
+
+#### Arguments
+
+| Argument Name | Description                                                                                                        |
+| ------------- | ------------------------------------------------------------------------------------------------------------------ |
+| command       | Specifies a command object. Refer to the [Write Side](write-side#sending-a-command) document for more information. |
```

Comment:
Why 'document'? Why not 'topic'? article?

Reply:
changed to 'article'